### PR TITLE
Update clap to v3.2

### PR DESCRIPTION
**Pull request by ducaale**
- Fix deprecation warnings
  - ~Replace `args_override_self` with `action` specified on each argument~
  - Replace `ArgEnum` with `ValueEnum`
  - Replace `parse()` with the new `value_parser`
  - implement `ValueParserFactory` for `Verify` See https://github.com/clap-rs/clap/discussions/4007
  - Introduce an enum for TLS version
- ~Bump MSRV to 1.56.1~

Resolves https://github.com/ducaale/xh/issues/262

### CODE REVIEW

**Comment by ducaale** [comment]
I am yet to find a replacement for `args_override_self` that works for negation flags
```
$ cargo run -q -- : --no-json --no-json
error: The argument '--no-json' was provided more than once, but cannot be used multiple times

USAGE:
    xh [OPTIONS] <[METHOD] URL> [--] [REQUEST_ITEM]...

For more information try --help
```

**Comment by ducaale** [comment]
Interesting, I wasn't aware that some options were un-deprecated.

I have also reverted to using the un-deprecated [`get_possible_values()`](https://docs.rs/clap/latest/clap/struct.Arg.html#method.get_possible_values) in `generate_manpages()` to ensure `[possible values: ...]` is generated for any `ArgEnum` that doesn't have an explicit name i.e `Theme`, `Pretty` and `AuthType`.

File: src/cli.rs [hunk 1]
```diff
@@ -563,12 +557,18 @@ impl Cli {
...
+        // XH_HTTPIE_COMPAT_MODE
+        if matches!(
+            matches.value_source("check-status"),
+            Some(clap::ValueSource::CommandLine)
+        ) || matches!(
+            matches.value_source("no-check-status"),
+            Some(clap::ValueSource::CommandLine)
+        ) {
+            self.check_status = Some(self.check_status_raw);
+        }
```
Comment (blyxxyz) [inline src/cli.rs:571]: Maybe it would be better to replace `is_present()` by `get_flag()`

**Review by blyxxyz — COMMENTED** [review]
Looks good!

- Can you regenerate the man page so we can scrutinize its diff?
- Do you know how much work it'll be to upgrade to v4 after this?

**Comment by ducaale** [comment]
>Do you know how much work it'll be to upgrade to v4 after this?

Apart from a lifetime error which I still need to figure out how to approach, upgrading to v4.0 should require minimal changes.

File: completions/_xh [hunk 2]
```diff
@@ -20,18 +20,19 @@ _xh() {
 colors\:"Apply syntax highlighting to output"
 format\:"Pretty-print json and sort headers"
 none\:"Disable both coloring and formatting"))' \
+'--format-options=[Set output formatting options. The only currently implemented option is json.indent]:FORMAT_OPTIONS: ' \
 '-s+[Output coloring style]:THEME:(auto solarized monokai fruity)' \
 '--style=[Output coloring style]:THEME:(auto solarized monokai fruity)' \
-'--response-charset=[Override the response encoding for terminal display purposes]:ENCODING: ' \
+'*--response-charset=[Override the response encoding for terminal display purposes]:ENCODING: ' \
```
Comment (blyxxyz) [inline completions/_xh:26]: It's probably fine but I wonder what's going on here.

**Review by blyxxyz — APPROVED** [review]
Looks great!

File: completions/_xh (hunk 2 above)
Comment (blyxxyz) [reply completions/_xh:26]: zsh still completes it so ¯\\_ (ツ)_/¯

**Pull request merged**